### PR TITLE
feat(dashaboard): new filter 'my groups'

### PR DESCRIPTION
**Pull request by stonebuzz**
Add new filter 'my groups' to dashboard

image

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | internal 21250

### CODE REVIEW

File: inc/dashboard/provider.class.php [hunk 1]
```diff
@@ -1368,6 +1378,54 @@ private static function getFiltersCriteria(string $table = "", array $apply_filt
...
+    * get groups of the current user
+    *
+    * @param integer $userID
+    * @return array
+    */
+    static function getMyGroups($userID) {
+      global $DB;
+
+      // from Item_Ticket::dropdownMyDevices()
+      $dbUtil = new DbUtils();
```
Comment (trasher) [inline inc/dashboard/provider.class.php:1391]:
```suggestion
```

File: inc/dashboard/provider.class.php [hunk 2]
```diff
@@ -1368,6 +1378,54 @@ private static function getFiltersCriteria(string $table = "", array $apply_filt
...
+            $groupTable => [
+               'FKEY' => [
+                  $groupTable => 'id',
+                  $groupUserTable => $groupFk,
+               ],
+            ],
+         ],
+         'WHERE' => [
+            $groupUserTable . '.users_id' => $userID,
+         ] + $dbUtil->getEntitiesRestrictCriteria(
```
Comment (trasher) [inline inc/dashboard/provider.class.php:1411]:
```suggestion
         ] + getEntitiesRestrictCriteria(
```

File: inc/dashboard/provider.class.php [hunk 3]
```diff
@@ -1368,6 +1378,54 @@ private static function getFiltersCriteria(string $table = "", array $apply_filt
...
+            '',
+            $_SESSION['glpiactive_entity'],
+            $_SESSION['glpiactive_entity_recursive']
+         )
+      ]);
+      if ($result->count() === 0) {
+         return [];
+      }
+      foreach ($result as $data) {
+         $a_groups                     = $dbUtil->getAncestorsOf("glpi_groups", $data["groups_id"]);
```
Comment (trasher) [inline inc/dashboard/provider.class.php:1422]:
```suggestion
         $a_groups                     = getAncestorsOf("glpi_groups", $data["groups_id"]);
```

File: inc/dashboard/provider.class.php [hunk 4]
```diff
@@ -50,6 +50,8 @@
 use Ticket_User;
 use Toolbox;
 use User;
+use DbUtils;
```
Comment (trasher) [inline inc/dashboard/provider.class.php:53]:
```suggestion
```

**Review by trasher — CHANGES_REQUESTED** [review]

**Review by orthagh — CHANGES_REQUESTED** [review]
you can greatly simplify (by removing the entire `getMyGroups`) as you always found the current user's groups in `$_SESSION['glpigroups']`

File: inc/dashboard/provider.class.php (hunk 4 above)
Comment (stonebuzz) [reply inc/dashboard/provider.class.php]: > you can greatly simplify (by removing the entire `getMyGroups`) as you always found the current user's groups in `$_SESSION['glpigroups']`

Done

Comment (stonebuzz) [reply inc/dashboard/provider.class.php]: change by $_SESSION['glpi_groups']

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**